spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java

```diff
@@ -252,6 +251,37 @@ private void deployStream(StreamDefinition stream, Map<String, String> cumulated
 		}
 	}
 
+	private ModuleType determineModuleType(ModuleDefinition moduleDefinition, int index, boolean hasNext) {
```

Comment:
I will change to `isFirst`. Let's continue the discussion about auto-completion offline. I would imagine the logic is a bit different anyways - considering, for example, you could have a processor OR sink following a pipe.